features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/ConvertToEvent.java

```diff
@@ -230,6 +245,39 @@ public static final EventBuilder toEventBuilder(SyslogMessage message, String sy
         return bldr;
     }
 
+    private static InetAddress resolveHostName(LocationAwareDnsLookupClient locationAwareDnsLookupClient,
+                                               Cache<HostNameWithLocationKey, String> dnsCache,
+                                               String location, String systemId, SyslogMessage message) {
+
+        if (LocationUtils.isDefaultLocationName(location)) {
+            return message.getHostAddress();
+        }
+        String hostName = message.getHostName();
+        if (Strings.isNullOrEmpty(hostName) ||
+                locationAwareDnsLookupClient == null || dnsCache == null) {
+            return null;
+        }
+        // Try to find the element in the cache first.
+        String hostIpAddress = dnsCache.getIfPresent(new HostNameWithLocationKey(hostName, location));
```

Comment:
Minor nit, you end up doing `new HostNameWithLocationKey(hostName, location)` twice if it's not in the cache.